Ray-Only.ipynb

```diff
@@ -69,39 +76,123 @@
+    "Month 1, Day 1.\n",
+    "1. Fails currently because worker nodes keep dying.\n",
+    "2. Success! Successful compaction for a days worth of parquets. Next is a month."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
     "# Read path\n",
     "read_path = 'raydev/metric=cluster_version'\n",
     "read_bucket_uri = f's3://{bucket}/{read_path}/year={year}/month={month}/day={day}'\n",
-    "\n",
+    "print(read_bucket_uri)\n",
     "\n",
     "# Write path\n",
     "write_path = 'raydev-write-demo/metric=cluster_version'\n",
-    "write_bucket_uri = f's3://{bucket}/{write_path}/year={year}/month={month}/day={day}'\n"
+    "write_bucket_uri = f's3://{bucket}/{write_path}/year={year}/month={month}/day={day}'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# S3FileSystem in PyArrow\n",
+    "fs_pyarrow = pq.S3FileSystem(endpoint_override=endpoint)\n",
+    "\n",
+    "# Reading parquet using Ray through filesystem\n",
+    "df = ray.data.read_parquet(paths=read_bucket_uri, filesystem=fs_pyarrow)\n",
+    "\n",
+    "# Writing back a single parquet file\n",
+    "df.repartition(1).write_parquet(path=write_bucket_uri, filesystem=fs_pyarrow)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Following block reads one days worth of parquet files and writes back a single parquet file\n",
-    "Month 1, Day 1.\n",
-    "Fails currently because worker nodes keep dying."
+    "Following cells try compaction for a whole month:\n",
+    "1. Reading from multiple directories seems to be not supported\n",
+    "2. Union of dataframes is supported, can be done before the compaction step. https://github.com/ray-project/ray/issues/24598\n",
+    "Status: Fails due to memory limits, cannot load a dataframe for more than several days worth of parquets"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Read path\n",
+    "read_path = 'raydev/metric=cluster_version'\n",
+    "read_bucket_uri = f's3://{bucket}/{read_path}/year={year}/month={month}/day={day}'\n",
+    "# Holds paths \n",
+    "read_bucket_multiple = []\n",
+    "\n",
+    "# Creating paths for every day in the month.\n",
+    "for i in range(2,32):\n",
+    "    if i < 10:\n",
+    "        day = '0'+str(i)\n",
+    "    else:\n",
+    "        day = str(i)\n",
+    "    \n",
+    "    read_bucket_multiple.append(f\"s3://{bucket}/{read_path}/year={year}/month={month}/day={day}\")\n",
+    "                           \n",
+    "print(read_bucket_multiple)\n",
+    "\n",
+    "\n",
+    "# Write path\n",
+    "write_path = 'raydev-write-demo/metric=cluster_version'\n",
```

Comment:
should we extract these values (`raydev-write-demo` and `cluster_version`) to the reusable variable definitions cell?